### DISCUSSION TITLE
Welcome to servicenow-powershell Discussions!

### DISCUSSION THREAD

**Discussion by gdbarron** [Snow-Shell/servicenow-powershell#122] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JLogan3o13** [comment]
I just downloaded the module, and have a question. New-ServiceNowSession fails for me, but gives only the most generic "You cannot call a method on a null-valued expression". Is there a way to better trace what is causing it to fail? I am doing a simple connection attempt, like so:

`New-ServiceNowSession -Url "wynnlvtest.service-now.com/side_door.do" -Credential $SNOWCreds`

**Reply by gdbarron** [reply]
Adding `-verbose` usually helps and is an easy way to start troubleshooting.  If you remove `/side_door.do`, does it work?  Can you provide the full error?

**Reply by JLogan3o13** [reply]
Thanks for the quick response. 

/side_door.do is their way of getting around their SSO injection; if I just try to hit their instance I get an unauthorized error. I spoke to the app owner, and it appears he has REST access severely restricted (even after I asked him). Once I get that in place I will try again and post back.

**Reply by gdbarron** [reply]
Ok, sounds good.  Keep in mind, from what I've seen, the rest api does not support sso.

**Comment by pheresparza** [comment]
Hi guys,

Have a quick question for you guys, I am scripting several action interacting with ServiceNow using this amazing module, such script will run each period of time, each time starting a new ServiceNow session, I looked for any command to force the close connection.

the real doubt here is how handle this module the current connections? are they being closed when console gets closed (or it will be when the task that run my script finish) or should I consider a way to close it manually (forced) ?

Trying to avoid impact SN instance with the number of conns my script is going to perform 

JFEH
